api/models/series_test.go

```diff
@@ -90,3 +92,103 @@ func TestJsonMarshal(t *testing.T) {
 		}
 	}
 }
+
+func TestSetTags(t *testing.T) {
+	cases := []struct {
+		in  Series
+		out map[string]string
+	}{
+		{
+			in: Series{},
+			out: map[string]string{
+				"name": "",
+			},
+		},
+		{
+			in: Series{
+				Target: "a",
+			},
+			out: map[string]string{
+				"name": "a",
+			},
+		},
+		{
+			in: Series{
+				Target: `a\b`,
+			},
+			out: map[string]string{
+				"name": `a\b`,
+			},
+		},
+		{
+			in: Series{
+				Target: "a;b=c;c=d",
+			},
+			out: map[string]string{
+				"name": "a",
+				"b":    "c",
+				"c":    "d",
+			},
+		},
+		{
+			in: Series{
+				Target: "a;biglongtagkeyhere=andithasabiglongtagvaluetoo;c=d",
+			},
+			out: map[string]string{
+				"name":              "a",
+				"biglongtagkeyhere": "andithasabiglongtagvaluetoo",
+				"c":                 "d",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		c.in.SetTags()
+		if !reflect.DeepEqual(c.out, c.in.Tags) {
+			t.Fatalf("SetTags incorrect\nexpected:%v\ngot:     %v\n", c.out, c.in.Tags)
+		}
+	}
+}
+
+func BenchmarkSetTags_00tags_00chars(b *testing.B) {
+	benchmarkSetTags(b, 0, 0, 0)
+}
+
+func BenchmarkSetTags_20tags_32chars(b *testing.B) {
+	benchmarkSetTags(b, 20, 32, 32)
+}
+
+func benchmarkSetTags(b *testing.B, numTags, tagKeyLength, tagValueLength int) {
+	in := Series{
+		Target: "my.metric.name",
+	}
+
+	for i := 0; i < numTags; i++ {
+		in.Target = in.Target + ";" + randString(tagKeyLength)
+		in.Target = in.Target + "="
+		in.Target = in.Target + randString(tagValueLength)
+	}
+
+	b.ReportAllocs()
```

Comment:
interesting. everywhere else we assume if the user wants this, they use the `-test.benchmem` flag.
i'm hesitant of introducing this here and creating inconsistency with other tests.
is there a clear benefit?

Reply:
Well, it allows benchmarks that know that allocations are something useful to benchmark to enable them independently of other benchmarks that may run that don't need them.  I can remove it though.

Reply:
i suppose it's a fair argument. but "allocations being useful to measure", doesn't that go for most benchmark code, other than perhaps very computationally expensive or io bound stuff maybe.
and i just use the alloc measures typically as a way to give color on changes in the time taken.

Reply:
Right. I like allocation benchmarks on certain code, because it's less subject to noisy neighbor with local benchmarks. I removed it in my last commit though.

Reply:
ok

Reply:
fyi re noisy neighbours i use cpu isolation.
see https://gist.github.com/Dieterbe/a52c95a9603507670eb39274544ee1a8